### ISSUE TITLE
Download toolchains from behind a proxy

### ISSUE THREAD

**Issue by Aaronepower** [rust-lang-nursery/rustup.rs#35]
I'm currently behind a proxy, and can't download any toolchains. How do you set up proxy settings with multirust?

**Comment by mkpankov** [comment]
@Aaronepower do you have `http_proxy` and `https_proxy` set up?

**Comment by Aaronepower** [comment]
I'm on Windows forgot to say. But I do also have those set up as env variables.

**Comment by kikokikok** [comment]
Hi,

Any news on this issue, I can't use multirust because of a enterprise proxy. Is there any workaround ?
Thanks !

**Comment by lilianmoraru** [comment]
Same issue here. I am on Ubuntu 15.10, 64-bit.
I have `http_proxy`, `https_proxy` and `ftp_proxy` set up.
`multirust` - the script, picks them up and the command `multirust update` works.
`multirust-rs` - the Rust application, doesn't work with the proxy.

```
info: updating existing install for 'beta'
downloading file from: 'https://static.rust-lang.org/dist/channel-rust-beta.sha256'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-beta.sha256' to '/home/uidr1070/.multirust/tmp/tfr6k8vxppx6ngcd_file' (Network: connection refused)
info: updating existing install for 'nightly'
downloading file from: 'https://static.rust-lang.org/dist/channel-rust-nightly.sha256'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-nightly.sha256' to '/home/uidr1070/.multirust/tmp/x07lpef1zc6s24vr_file' (Network: connection refused)
info: updating existing install for 'stable'
downloading file from: 'https://static.rust-lang.org/dist/channel-rust-stable.sha256'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-stable.sha256' to '/home/uidr1070/.multirust/tmp/pr1qo3_cbbj0tk5g_file' (Network: connection refused)

   beta update FAILED
nightly update FAILED
 stable update FAILED
```

**Comment by lilianmoraru** [comment]
Tried today `rustup` and the issue is still present.
I had issues installing it(`multirust-rs` was installing over `multirust` but it doesn't any more - another issue):
```sh
Continue? (Y/n) Y

error: toolchain 'nightly' is not installed
rustup: command failed: /tmp/tmp.pHwfFAkit0/rustup-setup
```
To solve the issue I had to run `multirust` uninstall command(I tried removing the toolchains, then the metadata, finally after removing `multirust` it worked).

After this, I tried to install `rustup`:
```sh
Continue? (Y/n) 
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-stable.toml.sha256' to '/home/uidr1070/.multirust/tmp/zo5drrtemgah7y1s_file' (Network: connection refused)
rustup: command failed: /tmp/tmp.sr3f0QCId8/rustup-setup
```

**Comment by xen0n** [comment]
Waiting on this too, but according to #344 this bug should be fixed shortly.

**Comment by alexcrichton** [comment]
This should now be done! With the switch to curl in https://github.com/rust-lang-nursery/rustup.rs/pull/434 rustup should naturally pick up the relevant env vars now.

**Comment by lilianmoraru** [comment]
I'll test when I go back to work and report the results.

**Comment by xen0n** [comment]
Just updated to latest rustup, proxies confirmed working!

**Comment by lilianmoraru** [comment]
I can confirm that the behavior seems to be the same as with cargo.
- Works through a proxy
- Doesn't work through WPAD

All tools have problems running in `cmd` because of WPAD, so I guess this is the expected behavior :)
